todoapp/src/components/Todo.vue

```diff
@@ -1,40 +1,210 @@
 <!-- eslint-disable vue/multi-word-component-names -->
-
 <script setup lang="ts">
-import type { TextString } from '../types/text'
+import type { TodoTypes } from '../types/text'
+import Incomplete from '../assets/eclipseblack.svg'
+import Complete from '../assets/completedTask.png'
 
 const props = defineProps<{
-  todos: TextString[]
-}>()
-const emits = defineEmits<{
-  (e: 'removeTodo', index: number): void
+  todos: TodoTypes[]
+  isMobile: boolean
 }>()
 
-function removeTodo(index: number) {
-  emits('removeTodo', index)
+function handleImportance(priority: number) {
+  if (priority === 0) {
+    return 'low'
+  } else if (priority === 1) {
+    return 'medium'
+  } else if (priority === 2) {
+    return 'high'
+  } else {
+    return ''
+  }
 }
-</script>
 
+function statusState(todo: TodoTypes) {
+  if (todo.status === false) {
+    todo.status = true
+  } else {
+    todo.status = false
+  }
+}
+</script>
 <template>
-  <li class="todo" v-for="(todo, index) in todos" :key="index">
-    {{ todo.text }}
-    <button class="remove-todo-btn" @click="removeTodo(index)">remove</button>
-  </li>
-</template>
+  <div v-for="(todo, index) in todos" :key="index">
+    <div class="todo" v-if="isMobile">
+      <div @click="statusState(todo)" class="status-container">
+        <div v-if="!todo.status">
+          <img class="status" :src="Incomplete" alt="incomplete" />
+        </div>
+        <div v-else>
+          <img class="status" :src="Complete" alt="complete" />
+        </div>
+      </div>
+      <div class="todo-importance-wrap">
+        <div class="todo-content">
+          <p class="todo-title">
+            {{ todo.title }}
+          </p>
+          <p class="todo-created_at">
+            {{ todo.created_at }}
+          </p>
+        </div>
+        <div class="todo-importance-container">
+          <p :class="handleImportance(todo.priority)" class="todo-importance"></p>
+        </div>
+      </div>
+    </div>
 
+    <div class="todo-desktop" v-else>
```

Comment:
Please try to have a single todo card for mobile and desktop without duplicating the code.
Try to resolve as much as you can from css and use if else only for part where is truly necessary